File: cw/python/credit_card_mask.py
# https://www.codewars.com/kata/5412509bd436bd33920011bc/train/python


import re

def maskify(cc):
    return cc if (len(cc) <= 4) else re.sub(r"[a-zA-Z0-9$]", "#", cc[:-4]) + cc[len(cc) - len(cc[:4]):]


if __name__ == "__main__":
    cc = input()
    print(maskify(cc))

File: codewars/rust/src/credit_card_mask.rs
// https://www.codewars.com/kata/5412509bd436bd33920011bc/train/rust

use std::io;

fn maskify(cc: &str) -> String {
	if cc.len() >= 5 {
		let pos = cc.len() - 4;
		format!("{}{}", "#".repeat(pos), &cc[pos..])
	} else {
		cc.to_string()
	}
}

fn main() {
	let mut cc = String::new();

	io::stdin()
		.read_line(&mut cc)
		.expect("Failed to read line");

	println!("{}", maskify(&cc));
}